File: app/src/main/java/com/fitaleks/heroesthesaurus/characters/CharactersContract.java
package com.fitaleks.heroesthesaurus.characters;

import com.fitaleks.heroesthesaurus.BasePresenter;
import com.fitaleks.heroesthesaurus.BaseView;
import com.fitaleks.heroesthesaurus.data.Character;

import java.util.List;

/**
 * Created by Alexander on 01.12.16.
 */

public interface CharactersContract {
    interface View extends BaseView<Presenter> {
        void setLoadingIndicator(boolean active);

        void showCharacters(List<Character> characters);
        void showLoadingError();
    }

    interface Presenter extends BasePresenter {
        void loadCharacters(boolean forceUpdate);
    }
}

File: app/src/main/java/com/fitaleks/heroesthesaurus/ui/CharactersContract.kt
package com.fitaleks.heroesthesaurus.ui

import com.fitaleks.heroesthesaurus.BasePresenter
import com.fitaleks.heroesthesaurus.BaseView
import com.fitaleks.heroesthesaurus.data.MarvelCharacter

/**
 * Created by Alexander on 01.12.16.
 */
interface CharactersContract {
    interface View : BaseView<Presenter> {
        fun setLoadingIndicator(active: Boolean)

        fun showCharacters(characters: List<MarvelCharacter>)
        fun showLoadingError()
    }

    interface Presenter : BasePresenter {
        fun loadCharacters(forceUpdate: Boolean)
    }
}
